components/script/dom/htmltextareaelement.rs

```diff
@@ -335,6 +335,14 @@ impl HTMLTextAreaElement {
     fn selection(&self) -> TextControlSelection<Self> {
         TextControlSelection::new(&self, &self.textinput)
     }
+
+    // Helper function to check if text_cursor is to be updated or not
+    fn update_text_contents(&self, update_text_cursor: bool) {
+        if !update_text_cursor {
```

Comment:
Same as the other helper method - we'll need to take the implementation of SetValue and move it here, and call update_text_contents from reset instead of SetValue.